akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/AccountExample5b.scala

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright (C) 2017-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package docs.akka.persistence.typed
+
+import akka.actor.typed.Behavior
+import akka.persistence.typed.scaladsl.Effect
+import akka.persistence.typed.scaladsl.PersistentBehaviors5
+
+/*
+API experiment with factory for command and event handler
+- same as AccountExample5 but with explicit types rather than type alias
+- no enclosing class
+- nothing special at all, using defs for different command handlers
+- using PersistentBehaviors5
+*/
+
+object AccountExample5b {
```

Comment:
My opinion is that this is what we should use. Plain basics. If it's considered too verbose and hiding the types is preferred it can be complemented with the factories that are illustrated in AccountExample4.

Reply:
Just an observation .. Typically in domain driven design, we consider entities as pure Java / Scala objects and have the various operations as domain services on the object. e.g.

```scala
// domain entity
case class Account(accountNo: String, name: String, dateOfOpening: Date, dateOfClosing: Option[Date], ...)
```

and have all operations as domain services ..

```scala
trait AccountService {
  def open(..): Option[Account]
  def close(..): ..
  def debit(..): ..
  //..
}
```
and have various implementations of the trait. Actually it can be parameterized on types to facilitate multiple implementations. But that's a diff issue.

The point is the separation between the domain aggregate and the services. The concerns of persistence have to be baked into the services I think. And we need to think of changing implementations for that.

Just another point of view ..